org.osgi.test.cases.serviceloader/src/org/osgi/test/cases/serviceloader/junit/ServiceLoaderServiceClientTest.java

```diff
@@ -213,7 +213,9 @@ public void testServiceClientWithFilter() throws Exception {
     				wires = rev.getWiring().getRequiredWires("osgi.serviceloader");
     				assertNotNull(wires);
 
-    				assertEquals("There are two providers so there should be two wires", 2, wires.size());
+					assertTrue(
```

Comment:
Can we do

```java
import static org.assertj.core.api.Assertions.assertThat;

assertThat(wires).as("There should be 2 wires as there are 2 providers, but 1 wire is also allowed since cardinality:=multiple is not guaranteed to provide all wires")
    .hasSizeBetween(1, 2);
```

This reads better and will give a more meaningful exception if the size is wrong.